openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -372,18 +377,24 @@ def _verify_user(request, userid_in_session):
                 # conditionally set the log level.
                 log_func = log.debug if request.user.id is None else log.warning
                 log_func(
-                    u"SafeCookieData user at request '{0}' does not match user at response: '{1}'".format(
-                        request.safe_cookie_verified_user_id,
-                        request.user.id,
+                    (
+                        "SafeCookieData user at request '{0}' does not match user at response: '{1}' "
+                        "for request path '{2}'"
+                    ).format(
+                        request.safe_cookie_verified_user_id, request.user.id, request.path,
                     ),
                 )
+                set_custom_attribute("safe_sessions.user_mismatch", "request-response-mismatch")
             if request.safe_cookie_verified_user_id != userid_in_session:
                 log.warning(
-                    u"SafeCookieData user at request '{0}' does not match user in session: '{1}'".format(  # pylint: disable=logging-format-interpolation
-                        request.safe_cookie_verified_user_id,
-                        userid_in_session,
+                    (
+                        "SafeCookieData user at request '{0}' does not match user in session: '{1}' "
+                        "for request path '{2}'"
+                    ).format(  # pylint: disable=logging-format-interpolation
+                        request.safe_cookie_verified_user_id, userid_in_session, request.path,
                     ),
                 )
+                set_custom_attribute("safe_sessions.user_mismatch", "request-session-mismatch")
```

Comment:
If you end up leaving this in and relying upon it for monitoring or something, you may want to add it to a unit test.